### PR TITLE
REF: use should_extension_dispatch for comparison method

### PR DESCRIPTION
**Pull request by jbrockmendel**
This sits on top of #27815

### CODE REVIEW

File: pandas/core/ops/__init__.py [hunk 1]
```diff
@@ -605,14 +604,20 @@ def dispatch_to_extension_op(op, left, right):
     new_right = extract_array(right, extract_numpy=True)
 
     try:
-        res_values = op(new_left, new_right)
+        with warnings.catch_warnings():
```
Comment (jreback) [inline pandas/core/ops/__init__.py]: can you add a comment on the condition on what this is filtering. can we pre-empt the DeprecationWarning by a check?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/ops/__init__.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/ops/__init__.py]: turns out this was the wrong/suboptimal place to do this check.  so this catch_warnings is now removed entirely and this is handled by checking for ABCExtensionArray before we get here

**Comment by jbrockmendel** [comment]
updated+green

**Review by TomAugspurger — APPROVED** [review]

**Comment by TomAugspurger** [comment]
Thanks!

**Pull request merged**